SimG4Core/Application/src/RunManagerMTWorker.cc

```diff
@@ -186,7 +186,7 @@ RunManagerMTWorker::~RunManagerMTWorker() {
 }
 
 void RunManagerMTWorker::resetTLS() {
-  delete m_tls;
+  //delete m_tls;
```

Comment:
Just curious, do we know everything "owned" by `m_tls` is now deleted by Geant4? Or can this change lead to memory leaks?

I'd suggest to change the objects deleted (i.e. owned) elsewhere changed from `unique_ptr` to raw pointer in `TLSData` in a future PR to make the ownership more clear.

Reply:
@makortel , In Geant4 10.6 Geant4 deletion end of run was substantially improved comparing with previous versions. The proposed fix is a temporary one. We need to change interface to sensitive detectors and not delete them in CMSSW. This should be done in next PRs, when this will be merged. Sensitive detectors cannot be owned by CMSSW. Other objects should be checked as well.

Reply:
Thanks for the clarifications!